### ISSUE TITLE
Error: Provider produced inconsistent result after apply

### ISSUE THREAD

**Issue by segator** [hashicorp/terraform-provider-vault#1695]
Hi there,

### Terraform Version
Terraform v1.3.6
on linux_amd64
+ provider registry.terraform.io/hashicorp/vault v3.11.0

### Vault Server
is an HCP Enterprise vault server, fully managed by hashicorp.
### Affected Resource(s)
Please list the resources as a list, for example:
- vault_identity_group_alias

### Terraform Configuration Files
```hcl
resource "vault_identity_group_alias" "roche_gh_action_group_alias" {
  for_each = {
    for obj in local.vault_external_group_github_action_list: "${obj.vault_external_group}_${obj.github_oidc_sub}" => obj
  }
  name           = each.value.github_oidc_sub
  mount_accessor = vault_jwt_auth_backend.github_actions_oidc.accessor
  canonical_id   = vault_identity_group.edgecoreservices_external_group[each.value.vault_external_group].id
}
```
### Panic Output
```bash
ault_identity_group_alias.roche_gh_action_group_alias["ci-infrastructure-sandbox_repo:roche-dia-ris-platform/edge-platform-infrastructure-modules:environment:pull_request"]: Creation complete after 0s [id=29a0ac1a-251f-c74e-a70b-fb42d88fb62a]
╷
│ Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to
│ vault_identity_group_alias.roche_gh_action_group_alias["ci-infrastructure-sandbox_repo:roche-dia-ris-platform/edge-platform-infrastructure:environment:sandbox"], provider
│ "provider[\"registry.terraform.io/hashicorp/vault\"]" produced an unexpected new value: Root resource was present, but now absent.
│ 
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
```

### Actual Behavior
It crashes randomly I need to execute tf apply multiple times until it works.

**Comment by fairclothjm** [comment]
Hello @segator and @trodemaster thanks for reporting this issue. I am sorry you are having trouble.

Do either of you have any minimum reproduction steps for this? That would be very helpful! I tried reproducing with:

```
provider "vault" {
  address = "http://localhost:8200"
}

terraform {
  required_providers {
    vault = {
      source  = "hashicorp/vault"
      version = "~> 3.23.0"
    }
  }
}

resource "vault_identity_group" "group" {
  for_each = toset( ["1", "2", "3", "4", "5", "6", "7"] )
  name = "group_${each.key}"
  type = "external"
  policies = ["test"]
}

resource "vault_auth_backend" "githubA" {
  type = "github"
  path = "githubA-group"
}

resource "vault_identity_group_alias" "group_alias" {
  for_each = vault_identity_group.group
  name           = "alias_${each.key}"
  mount_accessor = vault_auth_backend.githubA.accessor
  canonical_id   = each.value.id
}
```

Any tips would be appreciated, thanks!

**Comment by fairclothjm** [comment]
@trodemaster Yes, I thought that might be the case here given the resource dependency. Debug logs could be helpful in this case. I understand that might be difficult as well if this is running in Prod and intermittent.

**Comment by trodemaster** [comment]
Ok I found some time to create a reproduction in terraform code. We confirmed that this throws errors when run on our production cluster but works as expected on a single-node test vault instance on my laptop. We are looking into options for getting a debug log over to hashicorp support. 

```
# variables
variable "vault_addr" {
  description = "The address of the Vault server"
  type        = string
  default     = "http://127.0.0.1:8200"
}

variable "number_of_groups" {
  description = "The number of groups to create"
  type        = number
  default     = 30
}

# provider configuration
provider "vault" {
  address = var.vault_addr
  skip_tls_verify = true
}

# create a list of numbers to use in the for_each loop
locals {
  number_set = toset([for i in range(1, var.number_of_groups + 1) : format("%d", i)])
}

# specify the required providers
terraform {
  required_providers {
    vault = {
      source  = "hashicorp/vault"
      version = "~> 3.24.0"
    }
  }
}

# setup userpass auth method
resource "vault_auth_backend" "userpass" {
  type = "userpass"
  path = "userpass"
}

# Construct the vault policy doc
data "vault_policy_document" "test" {
  rule {
    path         = "/secret"
    capabilities = ["read"]
    description  = "Simple test policy"
  }
}

# create a vault policy from the document
resource "vault_policy" "test" {
  name   = "grp-vault-test"
  policy = data.vault_policy_document.test.hcl
}

# create the vault identity groups using the number of groups
resource "vault_identity_group" "group" {
  for_each = local.number_set
  name     = "grp-vault-test-${each.key}"
  type     = "external"
  policies = ["test"]
}

# create the group aliases
resource "vault_identity_group_alias" "group" {
  for_each       = vault_identity_group.group
  name           = "grp-vault-test-${each.key}"
  mount_accessor = vault_auth_backend.userpass.accessor
  canonical_id   = each.value.id
}
```

Again the exact error from this run.
```
Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to vault_identity_group_alias.group["15"], provider "provider[\"registry.terraform.io/hashicorp/vault\"]" produced an unexpected new value: Root object was present,
│ but now absent.
│ 
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
```

**Comment by fairclothjm** [comment]
closed by #2140
